modin/pandas/test/dataframe/test_binary.py

```diff
@@ -492,13 +513,40 @@ def test_non_commutative_multiply():
         pytest.param(3.5, id="float scalar"),
     ],
 )
-def test_arithmetic_with_tricky_dtypes(val1, val2, op):
+def test_arithmetic_with_tricky_dtypes(val1, val2, op, request):
     modin_df1, pandas_df1 = create_test_dfs(val1)
     modin_df2, pandas_df2 = (
         create_test_dfs(val2) if isinstance(val2, list) else (val2, val2)
     )
+
+    raising_exceptions = None
+    if (
+        "bool-bool" in request.node.callspec.id
+        or "bool scalar-bool" in request.node.callspec.id
```

Comment:
> I wonder if we are going to proceed using request.* to identify the exact errors and how difficult it is for developers to find the exact cases for those errors using request.*?

Well, this is quite inconvenient, but this is the price we need to pay in order to be more confident that what is actually being tested is what is intended.

I would also like to note that to avoid this inconvenience, when writing a test, we need to separate test cases that should work correctly from those that should not.

Reply:
> I would also like to note that to avoid this inconvenience, when writing a test, we need to separate test cases that should work correctly from those that should not.

Yes, separating tests would probably make sense here. Matching error-prone parameters with exceptions would help I think.